regression-test/suites/nereids_rules_p0/mv/union_rewrite/partition_curd_union_rewrite_hive.groovy

```diff
@@ -0,0 +1,287 @@
+            from ${catalog_name}.${db}.${lineitem_tb_name} as t1
+            left join ${catalog_name}.${db}.${orders_tb_name} as t2 on l_orderkey = o_orderkey and l_shipdate = o_orderdate
+            where (l_shipdate>= '2023-10-18' and l_shipdate <= '2023-10-19')
+            group by
+            l_shipdate,
+            o_orderdate,
+            l_partkey,
+            l_suppkey
+            """
+
+        sql """DROP MATERIALIZED VIEW IF EXISTS ${mv_name}"""
+        sql """DROP TABLE IF EXISTS ${mv_name}"""
+        sql """
+            CREATE MATERIALIZED VIEW ${mv_name} 
+            BUILD IMMEDIATE REFRESH AUTO ON MANUAL
+            partition by(l_shipdate)
+            DISTRIBUTED BY RANDOM BUCKETS 2
+            PROPERTIES ('replication_num' = '1') 
+            AS 
+            ${mv_def_sql}
+            """
+
+        def order_by_stmt = " order by 1,2,3,4,5"
+        waitingMTMVTaskFinished(getJobName(db, mv_name))
+
+        // All partition is valid, test query and rewrite by materialized view
+        explain {
+            sql("${all_partition_sql}")
+            contains("${mv_name}(${mv_name})")
+        }
+        compare_res(all_partition_sql + order_by_stmt)
+        explain {
+            sql("${partition_sql}")
+            contains("${mv_name}(${mv_name})")
+        }
+        compare_res(partition_sql + order_by_stmt)
+
+        // Part partition is invalid, test can not use partition 2023-10-17 to rewrite
+        sql """
+            insert into ${catalog_name}.${db}.${lineitem_tb_name} values 
+            (1, 2, 3, 4, 5.5, 6.5, 7.5, 8.5, 'o', 'k', '2023-10-17', '2023-10-17', 'a', 'b', 'yyyyyyyyy', '2023-10-17');
+            """
+        // wait partition is invalid
+        sleep(5000)
+        explain {
+            sql("${all_partition_sql}")
+            contains("${mv_name}(${mv_name})")
+        }
+        assertTrue(compare_res(all_partition_sql + order_by_stmt) == false)
+        explain {
+            sql("${partition_sql}")
+            contains("${mv_name}(${mv_name})")
+        }
+        assertTrue(compare_res(all_partition_sql + order_by_stmt) == false)
+
+
+        sql "REFRESH MATERIALIZED VIEW ${mv_name} AUTO"
+        waitingMTMVTaskFinished(getJobName(db, mv_name))
+        explain {
+            sql("${all_partition_sql}")
+            contains("${mv_name}(${mv_name})")
+        }
+        compare_res(all_partition_sql + order_by_stmt)
+
+        explain {
+            sql("${partition_sql}")
+            contains("${mv_name}(${mv_name})")
+        }
+        compare_res(partition_sql + order_by_stmt)
+
+
+        // Test when base table create partition
+        sql """
+            insert into ${catalog_name}.${db}.${lineitem_tb_name} values 
+            (1, 2, 3, 4, 5.5, 6.5, 7.5, 8.5, 'o', 'k', '2023-10-21', '2023-10-21', 'a', 'b', 'yyyyyyyyy', '2023-10-21');
+            """
+        // Wait partition is invalid
+        sleep(5000)
+        explain {
```

Comment:
Should use `mv_rewrite_success` in Suite.grovvy, to location not chosen mv reason easier

Reply:
All have been modified completely.